### ISSUE TITLE
Support Binary data formats

### ISSUE THREAD

**Issue by TDI-tenderholt** [Veritomyx/PeakInvestigator-Python-SDK#3]
PI 2.1 will support a new binary format. It's defined as:

```
# Optional header information
# As many lines as desired
$[ Number of pairs ] [...] [ \n ]
# checksum:[...]
```

with following characteristics:

* Header/checksum lines begin with # and the single data line begins with $.
* The data is zlib compressed (default compression level)
* The first 8 bytes of the data line is the number of data points (signed long, big/network endian).
* The remaining data line costs of X double pairs (big/network endian) followed by a single \n character.

Writing in this format should be done with a function with the following signature:

`save_binary(headers, data_points, handle)` where `headers` is a collection of text (e.g. list of strings), data_points can be iterated through (e.g. `for data_point in data_points:`) and each item has two entries (i.e. `data_point[0]` is the mass, `data_point[1]` is the height), and `handle` is a (file object)[https://docs.python.org/3/glossary.html#term-file-object]. `handle` should not be closed because it could be for an open network socket where multiple scans are being sent.

Test files should be attached.

**Comment by TDI-tenderholt** [comment]
[BinaryFormatTestFiles.zip](https://github.com/Veritomyx/PeakInvestigator-Python-SDK/files/1988807/BinaryFormatTestFiles.zip)

**Comment by TDI-tenderholt** [comment]
The `struct` module is probably required: https://docs.python.org/2/library/struct.html. There's an example of its usage Uploader: https://github.com/Veritomyx/PeakInvestigator-Python-SDK/blob/master/peakinvestigator/progress/uploader.py#L34.

The number of pairs should be a long (i.e. `struct.pack('>l', ...)`) and the individual datapoints should be double (i.e. `struct.pack('>d', ...)`).

**Comment by jct197** [comment]
Hi Adam,

Attached is the example.bin file I've used as comparison and the output
from posted code.

-John

**Comment by TDI-tenderholt** [comment]
Closed via #21 .
